llvm/include/llvm-c/CAS/PluginAPI_functions.h

```diff
@@ -0,0 +1,244 @@
+/*===-- llvm-c/CAS/PluginAPI_functions.h - CAS Plugin Functions Interface -===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* The functions for the LLVM CAS plugin API. Intended for assisting          *|
+|* implementations of the API.                                                *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_C_CAS_PLUGINAPI_FUNCTIONS_H
+#define LLVM_C_CAS_PLUGINAPI_FUNCTIONS_H
+
+#include "llvm-c/CAS/PluginAPI_types.h"
+#include "llvm-c/ExternC.h"
+
+#ifndef LLCASAPI_PUBLIC
+#ifdef _WIN32
+#define LLCASAPI_PUBLIC __declspec(dllexport)
+#else
+#define LLCASAPI_PUBLIC
+#endif
+#endif
+
+LLVM_C_EXTERN_C_BEGIN
+
+/**
+ * Returns the \c LLCASAPI_VERSION_MAJOR and \c LLCASAPI_VERSION_MINOR values
+ * that the plugin was compiled with.
+ * Intended for assisting compatibility with different versions.
+ */
+LLCASAPI_PUBLIC void llcasapi_get_plugin_version(unsigned *major,
+                                                 unsigned *minor);
+
+/**
+ * Releases memory of C string pointers provided by other functions.
+ */
+LLCASAPI_PUBLIC void llcasapi_string_dispose(char *);
+
+LLCASAPI_PUBLIC llcasapi_cas_options_t llcasapi_cas_options_create(void);
+
+LLCASAPI_PUBLIC void llcasapi_cas_options_dispose(llcasapi_cas_options_t);
+
+/**
+ * Receives the \c LLCASAPI_VERSION_MAJOR and \c LLCASAPI_VERSION_MINOR values
+ * that the client was compiled with.
+ * Intended for assisting compatibility with different versions.
+ */
+LLCASAPI_PUBLIC void
+llcasapi_cas_options_set_client_version(llcasapi_cas_options_t, unsigned major,
```

Comment:
Is this a `get` or a `set`?

Reply:
`set`, this is for the client, e.g. clang, to inform the plugin of its version. The associated "get" one is `llcasapi_get_plugin_version` where the client gets the same info from the plugin.